src/lib/users.ts

```diff
@@ -10,3 +16,107 @@ export async function loadCurrentUser(
 
   return user
 }
+
+export async function loadUserProgress(
+  user_id: number,
+  page = 1,
+  per_page = 5,
+  token?: string,
+): Promise<any> {
+  const query = gql`
```

Comment:
this query feels like it's way more beefy than it needs to be but I was just trying to get it done and on the page so we can start to consider it